src/sentry/api/endpoints/data_export.py

```diff
@@ -42,16 +42,26 @@ def post(self, request, organization):
         data = serializer.validated_data
 
         try:
-            # TODO(Leander): Prevent repeated requests for identical queries per organization, if one is in progress
-            data_export = ExportedData.objects.create(
+            existing_data_exports = ExportedData.objects.filter(
                 organization=organization,
                 user=request.user,
                 query_type=data["query_type"],
                 query_info=data["query_info"],
-            )
+                date_finished=None,
+            ).order_by("-date_added")
+            if len(existing_data_exports) > 0:
+                data_export = existing_data_exports[0]
+                status = 200
```

Comment:
I forgot I added this --

This will prevent users from making multiple jobs to get their data only if an identical job (same type, info, user and org) is incomplete.